### PR TITLE
Documentation/zenko 1578 bucket policy calls

### PR DESCRIPTION
**Pull request by wabernat**
This PR adds bucket policy API documentation to the reference guide. We need it because we need to document these new calls. It also reorganizes the reference guide in a way that makes more sense, and ideally makes the book and the product it documents make more sense. 

This fixes #ZENKO-1578 and then some. 

Sorry there's so much to go through.

### CODE REVIEW

File: docs/docsource/reference/access_controls/ACLs/index.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ structure).
 .. code::
 
    <?xml version="1.0" encoding="UTF-8"?>
-   <AccessControlPolicy xmlns="http://s3.scality.com/doc/2006-03-01/">
+   <AccessControlPolicy xmlns="http://example.com/doc/2006-03-01/">
```
Comment (tmacro) [inline docs/docsource/reference/access_controls/ACLs/index.rst:23]: :+1:

**Review by tmacro — APPROVED** [review]

**Comment by wabernat** [comment]
@bert-e approve

File: docs/docsource/operation/Zenko_CLI/index.rst [hunk 2]
```diff
@@ -31,4 +32,4 @@ or both.
    CRR_Pause_&_Resume
    Object_Lifecycle_Management
    Monitoring_NFS-SOFS_Ingestion
-
+   managing_bucket_policies
```
Comment (dora-korpar) [inline docs/docsource/operation/Zenko_CLI/index.rst:35]: Should this be capitalized?

Comment (wabernat) [reply docs/docsource/operation/Zenko_CLI/index.rst:35]: capitalization on file names is not consistent.

File: docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.. _managing bucket policies:
+
+Managing Bucket Policies
+========================
+
+In addition to controlling access using access control list (ACL) policies
```
Comment (dora-korpar) [inline docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: The proper term here would be access control lists (they're technically not policies) and they're not user-based :)

Comment (wabernat) [reply docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: Thanks! Will fix.

Comment (dora-korpar) [reply docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: I'd also group CORS in with ACLs at the beginning of the sentence, sth like "In addition to controlling access using access control lists (ACLs) and cross-origin resource sharing (CORS), bucket owners can control access at the resource level using bucket policies."

Comment (wabernat) [reply docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: done.

File: docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+(user-based), bucket owners can control access at the resource level using
+cross-origin resource sharing (CORS) policies and bucket policies. Using bucket
+policies, a bucket's owner can set policies that apply to all objects in a
```
Comment (dora-korpar) [inline docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: Policies can apply to the bucket itself or only specific objects within a bucket, too, not sure if that's worth noting

Comment (wabernat) [reply docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: noted...

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Bucket Policy
+=============
+
+Bucket policies afford you the opportunity to restrict or permit access to
+buckets based on policies owned by the account that owns a bucket. Account users
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst]: While it's technically correct, I feel like this first sentence is a little confusing. I think something like "...based on policies set on a bucket by the bucket owner account". The idea of policies being owned by the account could be a bit misleading

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+can create policies for specific buckets and then grant other accounts access
+permissions to the same bucket (and all objects in it).
+
+User accounts can create a bucket policy for any bucket they own, and can grant
+bucket access to other accounts. Once bucket access is granted, all accounts
+with shared access to the bucket can access all objects in the bucket. Bucket
+policies enable accounts (account root users) to restrict access to specific IP
+addresses and to specific HTTP referrers.
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst]: We're not supporting using IP addresses and HTTP referrers to control permissions yet :)

File: docs/docsource/reference/access_controls/bucket_policies/index.rst (hunk 5 above)
Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: yeah... I was kind of worried about this. The issue (and correct me if my understanding is wrong) is that there aren't users so much as accounts  in Zenko, so the account/user split is nerve-wracking to say the least. 

If I don't hear any more on this, I'll change it to "user".

File: docs/docsource/reference/access_controls/bucket_policies/index.rst (hunk 6 above)
Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: thanks!

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+=============
+
+Bucket policies afford you the opportunity to restrict or permit access to
+buckets based on policies owned by the account that owns a bucket. Account users
+can create policies for specific buckets and then grant other accounts access
+permissions to the same bucket (and all objects in it).
+
+User accounts can create a bucket policy for any bucket they own, and can grant
+bucket access to other accounts. Once bucket access is granted, all accounts
+with shared access to the bucket can access all objects in the bucket. Bucket
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst]: This isn't true. Bucket policies allow access only to the specific actions listed in them, so for example it's possible for one user to be granted access only a GetBucketLocation action, while another user could be granted GET access to any resource, while another user has access to everything.
We can discuss in person if needed :)

File: docs/docsource/reference/bucket_operations/delete_bucket_policy.rst [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+This DELETE operation uses the policy subresource to delete a specified bucket's
+policy. For any identity other than the root user of the account that owns the
+bucket, the identity must have the DeleteBucketPolicy permissions on the
+specified bucket and belong to the bucket owner's account to use this operation.
+
+In the absence of DeleteBucketPolicy permissions, S3 Connector returns a ``403
+Access Denied`` error. If the permissions are correct, but you are not using an
+identity that belongs to the bucket owner's account, S3 Connector returns a
+``405 Method Not Allowed`` error.
```
Comment (dora-korpar) [inline docs/docsource/reference/bucket_operations/delete_bucket_policy.rst:14]: 👌

File: docs/docsource/reference/bucket_operations/get_bucket_list_objects_v2.rst [hunk 9]
```diff
@@ -3,7 +3,7 @@
 GET Bucket (List Objects) Version 2
 ===================================
 
-The Version 2 implementation of the GET operation returns some or all (up to 
+The Version 2  GET operation returns some or all (up to
```
Comment (dora-korpar) [inline docs/docsource/reference/bucket_operations/get_bucket_list_objects_v2.rst]: extra space after 2?

File: docs/docsource/reference/access_controls/bucket_policies/index.rst (hunk 5 above)
Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: Strategic vagueness to the rescue: 

"based on policies set by a bucket's owner."

File: docs/docsource/reference/access_controls/bucket_policies/index.rst (hunk 6 above)
Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: done.

File: docs/docsource/reference/bucket_operations/get_bucket_list_objects_v2.rst (hunk 9 above)
Comment (wabernat) [reply docs/docsource/reference/bucket_operations/get_bucket_list_objects_v2.rst]: non-critical, but I will squish it.

**Comment by ghost** [comment]
There were the following issues with this Pull Request

* Commit: 6cf3b8a92fb9e3c360edc190e29d79d5a08ea048
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

File: docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst [hunk 10]
```diff
@@ -0,0 +1,19 @@
+.. _managing bucket policies:
+
+Managing Bucket Policies
+========================
+
+In addition to controlling user or process access using access control lists
+(ACLs) and cross-origin resource sharing (CORS), bucket owners can control
+access at the resource level using bucket policies. Using bucket policies, a
+bucket's owner can set policies that apply to all objects in a bucket or to
```
Comment (dora-korpar) [inline docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: They can also apply to just the buckets themselves. They don't necessarily apply to objects at all :)

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 11]
```diff
@@ -0,0 +1,49 @@
+Bucket Policy
+=============
+
+Bucket policies let owners inscribe highly specific or powerfully general access
+controls to buckets and the objects they contain. By default, external accounts
+are denied access to buckets. A bucket policy, set as a JSON file attached to
+the bucket, can permit or deny specified accounts or users access to the bucket,
+to all objects in the bucket, or to individual objects in the bucket, and can
+permit or deny specific actions on specified buckets or objects.
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst:9]: 👍

File: docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst (hunk 10 above)
Comment (wabernat) [reply docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst]: fixed.

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+Policies consist of three elements: a principal (the user, account, service,
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst]: there are actually 4 elements - I didn't mention the Effect key, which is just set to either "Allow" or "Deny".
Also, services and roles as principals are not either not supported by us yet (services) or not supported by AWS (roles)

File: docs/docsource/reference/access_controls/bucket_policies/index.rst [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+role, or entity being permitted or denied access), a resource (the bucket or
+object to which access is being granted or denied), and an action (the API being
+permitted or blocked for the principal or resource).
+
+The principal is defined by Amazon Resource Name (ARN), federated user, IAM
```
Comment (dora-korpar) [inline docs/docsource/reference/access_controls/bucket_policies/index.rst]: Similar note here. The principals we currently support are Account Id, Account Arn, User Arn, and Canonical Id. No federated users or roles

Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: fixed (I hope)

File: docs/docsource/reference/access_controls/bucket_policies/index.rst (hunk 12 above)
Comment (wabernat) [reply docs/docsource/reference/access_controls/bucket_policies/index.rst]: should be good now.

File: docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+Managing Bucket Policies
+========================
+
+In addition to controlling user or process access using access control lists
+(ACLs) and cross-origin resource sharing (CORS), bucket owners can control
+access at the resource level using bucket policies. Using bucket policies, a
+bucket's owner can set policies that apply to the bucket, to all objects in a
+bucket, or to specific objects in a bucket. Bucket policies provide added
+granularity to permission management.
```
Comment (dora-korpar) [inline docs/docsource/operation/Zenko_CLI/managing_bucket_policies.rst:11]: *chef's kiss*

**Review by dora-korpar — APPROVED** [review]

**Pull request merged**